src/Interpreters/HashJoin.cpp

```diff
@@ -1232,8 +1232,8 @@ DataTypePtr HashJoin::joinGetCheckAndGetReturnType(const DataTypes & data_types,
     {
         const auto & left_type_origin = data_types[i];
         const auto & [c2, right_type_origin, right_name] = right_table_keys.safeGetByPosition(i);
-        auto left_type = removeNullable(left_type_origin);
-        auto right_type = removeNullable(right_type_origin);
+        auto left_type = removeLowCardinality(removeNullable(left_type_origin));
```

Comment:
Can we use `recursiveRemoveLowCardinality` here like in type check for regular join?

https://github.com/ClickHouse/ClickHouse/blob/427e5f295cf0f88fc1ea0de2722f016d0998cfc7/src/Interpreters/join_common.cpp#L233

Reply:
Ah, yes.